src/test/scala/cognite/spark/v1/DataModelInstancesRelationTest.scala

```diff
@@ -101,65 +151,119 @@ class DataModelInstancesRelationTest extends FlatSpec with Matchers with SparkTe
       .save
 
   it should "ingest data" in {
-    insertRows(
-      primitiveExtId,
-      spark
-        .sql(
-          s"""select 2.0 as prop_float,
-             |true as prop_bool,
-             |'abc' as prop_string,
-             |'prim_test' as externalId""".stripMargin))
-    byExternalId(primitiveExtId, "prim_test") shouldBe "prim_test"
-    getNumberOfRowsUpserted(primitiveExtId, "datamodelinstances") shouldBe 1
-    bluefieldAlphaClient.dataModelInstances.deleteByExternalId("prim_test").unsafeRunSync()
+    val randomId = "prim_test_" + shortRandomString()
+    try {
+      retryWhile[Boolean](
+        {
+          Try {
+            insertRows(
+              primitiveExtId,
+              spark
+                .sql(s"""select 2.0 as prop_float,
+                  |true as prop_bool,
+                  |'abc' as prop_string,
+                  |'${randomId}' as externalId""".stripMargin)
+            )
+          }.isFailure
```

Comment:
maybe also log the error here, because it hides the reason? sth like
```suggestion
    try {
      retryWhile[Boolean](
        {
          val try = Try {
            insertRows(
              primitiveExtId,
              spark
                .sql(s"""select 2.0 as prop_float,
                  |true as prop_bool,
                  |'abc' as prop_string,
                  |'${randomId}' as externalId""".stripMargin)
            )
          }
          try.recover({case e => logger.info(e.getLocalizedMessage + e.getCause + e.getMessage)})
          try.isFailure
```

Reply:
It helps seeing the error when debugging :P

Reply:
I can't make it work with getLogger, doesn't seem like we use it in any tests so far, guess we just have to print it if debug 😁